annotator/teed/__init__.py

```diff
@@ -0,0 +1,55 @@
+"""
+Hello, welcome on board,
+"""
+from __future__ import print_function
+
+import os
+import cv2
+import numpy as np
+
+import torch
+
+from annotator.teed.ted import TED  # TEED architecture
+from einops import rearrange
+from modules import devices
+from annotator.util import load_model,safe_step
+from annotator.annotator_path import models_path
+
+class TEEDDector:
+    """https://github.com/xavysp/TEED"""
+
+    model_dir = os.path.join(models_path, "TEED")
+
+    def __init__(self):
+        self.device = devices.get_device_for("controlnet")
+        self.model = TED().to(self.device).eval()
+        remote_url = os.environ.get(
+            "CONTROLNET_TEED_MODEL_URL",
+            "https://huggingface.co/bdsqlsz/qinglong_controlnet-lllite/resolve/main/Annotators/7_model.pth",
+        )
+        model_path = load_model(
```

Comment:
I get following error:
```
***
    Traceback (most recent call last):
      File "D:\stable-diffusion-webui\modules\safe.py", line 137, in load_with_extra
        check_pt(filename, extra_handler)
      File "D:\stable-diffusion-webui\modules\safe.py", line 84, in check_pt
        check_zip_filenames(filename, z.namelist())
      File "D:\stable-diffusion-webui\modules\safe.py", line 76, in check_zip_filenames
        raise Exception(f"bad file inside {filename}: {name}")
    Exception: bad file inside D:\stable-diffusion-webui\extensions\sd-webui-controlnet\annotator\downloads\TEED\7_model.pth: 7_model/byteorder
```

Seems like A1111's strict safety check again:
```python
# Regular expression that accepts 'dirname/version', 'dirname/data.pkl', and 'dirname/data/<number>'
allowed_zip_names_re = re.compile(r"^([^/]+)/((data/\d+)|version|(data\.pkl))$")
data_pkl_re = re.compile(r"^([^/]+)/data\.pkl$")

def check_zip_filenames(filename, names):
    for name in names:
        if allowed_zip_names_re.match(name):
            continue

        raise Exception(f"bad file inside {filename}: {name}")
```

Reply:
I change pth model for A1111's strict safety check.
it can work in linux now.
just need to delete donwloaded old model